coreneuron/nrniv/main1.cpp

```diff
@@ -378,11 +378,13 @@ extern "C" void mk_mech_init(int argc, char** argv) {
     #endif
     // read command line parameters and parameter config files
 
-    corenrn_param.parse(argc, argv);
+    try {corenrn_param.parse(argc, argv);} catch (...) {exit(1);}
```

Comment:
The addition of exceptions is pretty nice! :+1: 
I would only suggest to use [`nrn_abort()`](https://github.com/BlueBrain/CoreNeuron/blob/f7d78fb83197d946f506ecaedbfafb8a64d362ac/coreneuron/nrnmpi/nrnmpi.cpp#L192) instead of `exit()` to make sure that all the MPI processes exit and also to be a bit cleaner since `nrnmpi_init()` is already called. 
It would also be nice to reformat this line like this:
```suggestion
try {
   corenrn_param.parse(argc, argv);
}
catch (...) {
   nrn_abort(1);
}
```

Reply:
Nice "Catch", Ioannis! See? I made a nerdy joke! hahaha, that was funny! Anyway, if `nrn_abort(1)` is functionally equivalent to `exit(1)` it's definitely the better option.

Reply:
Hahaha noooo :laughing: 
It should be the same functionality, yes. In the implementation of `MPI_Abort()` there is probably going to be called `exit()` at some point but I think that it is better to use the `nrn_<mpi_function>()` functions that call the related `MPI functions` for the MPI operations. You can read more about  `MPI_Abort()` [here](https://www.mpich.org/static/docs/v3.1/www3/MPI_Abort.html) and [here](https://stackoverflow.com/questions/54844522/mpi-abort-vs-exit).

Reply:
Should be `nrn_abort()` then. I'll check it out. Thanks!